iceoryx2-cal/conformance-tests/src/arc_sync_policy_trait.rs

```diff
@@ -10,31 +10,35 @@
 //
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 
-#[generic_tests::define]
-mod arc_sync_policy {
+use iceoryx2_bb_conformance_test_macros::conformance_test_module;
+
+#[allow(clippy::module_inception)]
+#[conformance_test_module]
+pub mod arc_sync_policy_trait {
```

Comment:
In order to create unique and predictive names for the declarative macros, which are generated by the proc macros, we can either create a new module with the same name as the containing module or a new module with the name of the containing module and a '_test' suffix. The former is easier to implement and  potentially also a bit less magic when the names just have to be equal.

Unfortunately, it is not possible obtain the file name from within the proc macro and use that name to create the macro name.